### DISCUSSION TITLE
rdk_doc/Quick_start/hardware_introduction/rdk_x5/

### DISCUSSION THREAD

**Discussion by giscus** [D-Robotics/rdk_doc#41] [General]
# rdk_doc/Quick_start/hardware_introduction/rdk_x5/

接口总览

https://d-robotics.github.io/rdk_doc/Quick_start/hardware_introduction/rdk_x5/

**Comment by kyosuke638** [comment]
I connected the power via Type-C at 5V, but it does not start. The green light is on, but the orange light does not illuminate. What should I do?

**Reply by xuguosheng110** [reply]
dear developer，do you install os on your sd card? and you can use uart interface to watch the log detail and put it here

**Reply by kyosuke638** [reply]
I downloaded `ubuntu-preinstalled-desktop-arm64.img` from https://archive.d-robotics.cc/downloads/os_images/ and flashed it onto an SD card using BalenaEtcher.

The boot log obtained via UART is as follows:

```
SNOTICE:  Welcome to Horizon X5 ASIC BOOTROM - V4.1
NOTICE:  		OTP config:
NOTICE:  			otp exist: true
NOTICE:  			test region size: 304
NOTICE:  			secure region size: 120
NOTICE:  			none secure region size: 56
NOTICE:   Enable MMU 

NOTICE:  Booting Trusted Firmware
NOTICE:  BL1: v2.8(release):
NOTICE:  BL1: Built : 17:42:12, Oct 19 2023
NOTICE:  Enter SPI NAND-FLASH Mode......
NOTICE:   Nand PAGE_SIZE_2K 

NOTICE:   Nand is_dummy = true 

NOTICE:   Nand CLK_12M 

NOTICE:  SPI baudrate is set to 12000000Hz
NOTICE:   Nand reset 

NOTICE:  reg_status should = 0x18
NOTICE:  Invalid enable flash buffer mode
NOTICE:  Manufacturer ID 0xc8  Device ID 0x41c8 

NOTICE:  Enter media_source_select process(5).
NOTICE:  SPI Nand flash set quad PASS
NOTICE:  Get S1PK
NOTICE:  [verify_signature] pass.
NOTICE:  [verify_hash] pass.
NOTICE:  [verify_hash] pass.
NOTICE:  [auth_decrypt] pass.
NOTICE:  BL1: BL2 memory layout address = 0x1fe9f000
NOTICE:  BL1: Booting BL2
E
NOTICE:  		OTP config:
NOTICE:  			otp exist: true
NOTICE:  			test region size: 304
NOTICE:  			secure region size: 120
NOTICE:  			none secure region size: 56
NOTICE:  origin vdd cpu:800mv, now vdd cpu: 850mv
NOTICE:  BL2: v2.8(release):v1.0.8-45-gc4d99b7e2
NOTICE:  BL2: Built : 09:51:10, Aug 28 2024
NOTICE:  Enter SPI NAND-FLASH Mode......
NOTICE:  dw_apb_ssi_params.apb_bus_freq: 100000000
NOTICE:  SPI baudrate is set to 12500000Hz
NOTICE:  Enter media_source_select process(5).
NOTICE:  BL2 config file
NOTICE:  ab slot is : 0
NOTICE:  fip index:0
NOTICE:  start to load bl2 cfg image
NOTICE:  BL2 CFG ADDR:0x1ff00000, MAGIC:0x474643324c424248
NOTICE:  bypass update efuse
NOTICE:  calibration_offset -9
NOTICE:  trimming_value 9
NOTICE:  ADC read channel[2]: 302mv
NOTICE:  ddr info:[LPDDR4]-[DUAL_RANK]-[ECC OFF]-[DVFS ON]-[4266M]
NOTICE:  JTAG ENABLE
NOTICE:  disable wdt
NOTICE:  Get S1PK
NOTICE:  [verify_signature] pass.
NOTICE:  [verify_hash] pass.
NOTICE:  [verify_signature] pass.
NOTICE:  start to load ddr image
NOTICE:  [verify_hash] pass.
NOTICE:  run ddr Fw
…NOTICE:  matching profile: profile_2 from aon config
NOTICE:  type: LP4, freq: 4266, ecc: 0(0x7f), rank: 2, dvfs: 1, version: 2.02
NOTICE:  ddr cost: 153793 us, ddr size: 8 GB, manuid: 0xff
NOTICE:  ddr ready
NOTICE:  Get S1PK
NOTICE:  [verify_signature] pass.
NOTICE:  [verify_hash] pass.
NOTICE:  [verify_signature] pass.
NOTICE:  start to load bl31 image
NOTICE:  [verify_hash] pass.
NOTICE:  Get S1PK
NOTICE:  [verify_signature] pass.
NOTICE:  [verify_hash] pass.
NOTICE:  [verify_signature] pass.
NOTICE:  start to load optee image
NOTICE:  [verify_hash] pass.
NOTICE:  [verify_hash] pass.
NOTICE:  start to load uboot image
NOTICE:  BL1: Booting BL31
OTICE:  multicore_init: sec_entrypoint = 0x20000108
NOTICE:  BL31: v2.8(release):v1.0.8-43-g09714a8e6
NOTICE:  BL31: Built : 19:17:06, Aug 16 2024
NOTICE:  plat_setup_psci_ops: sec_entrypoint = 0x20000108
I/TC: 
I/TC: OP-TEE version: 4c4cc7275 (gcc version 11.3.1 20220712 (Arm GNU Toolchain 11.3.Rel1)) #1 Mon Sep  2 09:27:57 UTC 2024 aarch64
I/TC: WARNING: This OP-TEE configuration might be insecure!
I/TC: WARNING: Please check https://optee.readthedocs.io/en/latest/architecture/porting_guidelines.html
I/TC: Primary CPU initializing
MBEDTLS_CORE  Select regular OTP
MBEDTLS_CORE  SRAM Pool Base: 0x80000000, size: 0x2000, alignment: 0x10
MBEDTLS_CORE   ACA driver init success!
MBEDTLS_CORE  set sw_init_done!
I/TC: Primary CPU switching to normal world boot


U-Boot 2022.10-g505104e8-dirty (Sep 26 2024 - 15:01:37 +0800)

Model: Horizon X5
DRAM:  7.9 GiB
console out start addr:0x0000000087ffc000, end addr:0x0000000087ffe000
horizon_lsio_pinctrl lsio_iomuxc@34180000: Initialized D-Robotics pinctrl driver
optee optee: OP-TEE: revision 3.20 (4c4cc727)
ADC calibration: -9
ADC trimming: 9
horizon_hsio_pinctrl hsio_iomuxc@35050000: Initialized D-Robotics pinctrl driver
Core:  126 devices, 23 uclasses, devicetree: separate
WDT:   Started watchdog@34250000 with servicing (10s timeout)
stop watchdog
MMC:   sdio-host@35040000: 0, sdio-host@35020000: 1
Loading Environment from Remote... *** Warning - bad CRC, using default environment

In:    serial@32120000
Out:   serial@32120000
Err:   serial@32120000
Net:   
Warning: gmac-tsn@35010000 (eth0) using random MAC address - 62:7a:18:04:b6:93
eth0: gmac-tsn@35010000
Hit <SPACE> key to stop autoboot in  2s
switch to partitions #0, OK
mmc1 is current device
SD Card detected,into ums update system
argv[1] 0
button no press,pass
Card did not respond to voltage select! : -110
eMMC or SD Card not detected on mmchost 0
switch to partitions #0, OK
mmc1 is current device
eMMC or SD Card detected on mmchost 1
Scanning mmc 1:2...
Found U-Boot script /boot/boot.scr
1064 bytes read in 18 ms (57.6 KiB/s)
## Executing script at 84100000
Boot script loaded from devtype:mmc devnum:1 devplist:2
bootargs = console=ttyS0,921600 video=hobot:x3sdb-hdmi rootfstype=ext4 rw rootwait root=/dev/mmcblk1p2 ubi.mtd=2,2048 mtdparts=hr_nand.0:6291456@0x0(miniboot),2097152@0x600000(env)
Loading fdt file: /boot/hobot/
Failed to load '/boot/hobot/'
Apply device tree overlay
** File not found /boot/config.txt **
DTOVERLAY[error]: can't load config file(/boot/config.txt)

Loading kernel: /boot/Image
19572744 bytes read in 921 ms (20.3 MiB/s)
Boot kernel from 0x85800000, devices tree from 0x84000000
ERROR: Did not find a cmdline Flattened Device Tree
Could not find a valid device tree
SCRIPT FAILED: continuing...
MMC Device 2 not found
no mmc device at slot 2
eMMC or SD Card not detected on mmchost 2
Hobot>
```

**Reply by xuguosheng110** [reply]
you can change some other better micro sd card， and install os, after that, upgrade your system ,(to upgrade miniboot on your nanflash) finally, The earliest SD card can be used now.

**Reply by kyosuke638** [reply]
How can I upgrade?

**Comment by ericlinwk** [comment]
I have same problem,
I try "ubuntu-preinstalled-desktop-arm64.img" and "ubuntu-preinstalled-server-arm64.img", got the same error. My board is X5.

Hobot>SNOTICE:  Welcome to Horizon X5 ASIC BOOTROM - V4.1
NOTICE:                 OTP config:
NOTICE:                         otp exist: true
NOTICE:                         test region size: 304
NOTICE:                         secure region size: 120
NOTICE:                         none secure region size: 56
NOTICE:   Enable MMU
NOTICE:  Booting Trusted Firmware
NOTICE:  BL1: v2.8(release):
NOTICE:  BL1: Built : 17:42:12, Oct 19 2023
NOTICE:  Enter SPI NAND-FLASH Mode......
NOTICE:   Nand PAGE_SIZE_2K
NOTICE:   Nand is_dummy = true
NOTICE:   Nand CLK_12M
NOTICE:  SPI baudrate is set to 12000000Hz
NOTICE:   Nand reset
NOTICE:  reg_status should = 0x18
NOTICE:  Invalid enable flash buffer mode
NOTICE:  Manufacturer ID 0xc8  Device ID 0x41c8
NOTICE:  Enter media_source_select process(5).
NOTICE:  SPI Nand flash set quad PASS
NOTICE:  Get S1PK
NOTICE:  [verify_signature] pass.
NOTICE:  [verify_hash] pass.
NOTICE:  [verify_hash] pass.
NOTICE:  [auth_decrypt] pass.
NOTICE:  BL1: BL2 memory layout address = 0x1fe9f000
NOTICE:  BL1: Booting BL2
NOTICE:                 OTP config:
NOTICE:                         otp exist: true
NOTICE:                         test region size: 304
NOTICE:                         secure region size: 120
NOTICE:                         none secure region size: 56
NOTICE:  origin vdd cpu:800mv, now vdd cpu: 850mv
NOTICE:  BL2: v2.8(release):v1.0.8-45-gc4d99b7e2
NOTICE:  BL2: Built : 09:51:10, Aug 28 2024
NOTICE:  Enter SPI NAND-FLASH Mode......
NOTICE:  dw_apb_ssi_params.apb_bus_freq: 100000000
NOTICE:  SPI baudrate is set to 12500000Hz
NOTICE:  Enter media_source_select process(5).
NOTICE:  BL2 config file
NOTICE:  ab slot is : 0
NOTICE:  fip index:0
NOTICE:  start to load bl2 cfg image
NOTICE:  BL2 CFG ADDR:0x1ff00000, MAGIC:0x474643324c424248
NOTICE:  bypass update efuse
NOTICE:  calibration_offset 10
NOTICE:  trimming_value 8
NOTICE:  ADC read channel[2]: 293mv
NOTICE:  ddr info:[LPDDR4]-[DUAL_RANK]-[ECC OFF]-[DVFS ON]-[4266M]
NOTICE:  JTAG ENABLE
NOTICE:  disable wdt
NOTICE:  Get S1PK
NOTICE:  [verify_signature] pass.
NOTICE:  [verify_hash] pass.
NOTICE:  [verify_signature] pass.
NOTICE:  start to load ddr image
NOTICE:  [verify_hash] pass.
▒NOTICE:  matching profile: profile_2 from aon config
NOTICE:  type: LP4, freq: 4266, ecc: 0(0x7f), rank: 2, dvfs: 1, version: 2.02
NOTICE:  ddr cost: 153709 us, ddr size: 8 GB, manuid: 0xff
NOTICE:  ddr ready
NOTICE:  Get S1PK
NOTICE:  [verify_signature] pass.
NOTICE:  [verify_hash] pass.
NOTICE:  [verify_signature] pass.
NOTICE:  start to load bl31 image
NOTICE:  [verify_hash] pass.
NOTICE:  Get S1PK
NOTICE:  [verify_signature] pass.
NOTICE:  [verify_hash] pass.
NOTICE:  [verify_signature] pass.
NOTICE:  start to load optee image
NOTICE:  [verify_hash] pass.
NOTICE:  [verify_hash] pass.
NOTICE:  start to load uboot image
▒NOTICE:  multicore_init: sec_entrypoint = 0x20000108
NOTICE:  BL31: v2.8(release):v1.0.8-43-g09714a8e6
NOTICE:  BL31: Built : 19:17:06, Aug 16 2024
NOTICE:  plat_setup_psci_ops: sec_entrypoint = 0x20000108
I/TC:
I/TC: OP-TEE version: 4c4cc7275 (gcc version 11.3.1 20220712 (Arm GNU Toolchain 11.3.Rel1)) #1 Mon Sep  2 09:27:57 UTC 2024 aarch64
I/TC: WARNING: This OP-TEE configuration might be insecure!
I/TC: WARNING: Please check https://optee.readthedocs.io/en/latest/architecture/porting_guidelines.html
I/TC: Primary CPU initializing
MBEDTLS_CORE  Select regular OTP
MBEDTLS_CORE  SRAM Pool Base: 0x80000000, size: 0x2000, alignment: 0x10
MBEDTLS_CORE   ACA driver init success!
MBEDTLS_CORE  set sw_init_done!
I/TC: Primary CPU switching to normal world boot


U-Boot 2022.10-g505104e8-dirty (Sep 26 2024 - 15:01:37 +0800)

Model: Horizon X5
DRAM:  7.9 GiB
console out start addr:0x0000000087ffc000, end addr:0x0000000087ffe000
horizon_lsio_pinctrl lsio_iomuxc@34180000: Initialized D-Robotics pinctrl driver
optee optee: OP-TEE: revision 3.20 (4c4cc727)
ADC calibration: 10
ADC trimming: 8
horizon_hsio_pinctrl hsio_iomuxc@35050000: Initialized D-Robotics pinctrl driver
Core:  126 devices, 23 uclasses, devicetree: separate
WDT:   Started watchdog@34250000 with servicing (10s timeout)
stop watchdog
MMC:   sdio-host@35040000: 0, sdio-host@35020000: 1
Loading Environment from Remote... *** Warning - bad CRC, using default environment

In:    serial@32120000
Out:   serial@32120000
Err:   serial@32120000
Net:
Warning: gmac-tsn@35010000 (eth0) using random MAC address - 0a:67:99:c1:6d:3f
eth0: gmac-tsn@35010000
Hit <SPACE> key to stop autoboot in  2s
switch to partitions #0, OK
mmc1 is current device
SD Card detected,into ums update system
argv[1] 0
button no press,pass
Card did not respond to voltage select! : -110
eMMC or SD Card not detected on mmchost 0
switch to partitions #0, OK
mmc1 is current device
eMMC or SD Card detected on mmchost 1
Scanning mmc 1:2...
Found U-Boot script /boot/boot.scr
1064 bytes read in 18 ms (57.6 KiB/s)
## Executing script at 84100000
Boot script loaded from devtype:mmc devnum:1 devplist:2
bootargs = console=ttyS0,921600 video=hobot:x3sdb-hdmi rootfstype=ext4 rw rootwait root=/dev/mmcblk1p2 ubi.mtd=2,2048 mtdparts=hr_nand.0:6291456@0x0(miniboot),2097152@0x600000(env)
Loading fdt file: /boot/hobot/
Failed to load '/boot/hobot/'
Apply device tree overlay
** File not found /boot/config.txt **
DTOVERLAY[error]: can't load config file(/boot/config.txt)

Loading kernel: /boot/Image
19572744 bytes read in 917 ms (20.4 MiB/s)
Boot kernel from 0x85800000, devices tree from 0x84000000
ERROR: Did not find a cmdline Flattened Device Tree
Could not find a valid device tree
SCRIPT FAILED: continuing...
MMC Device 2 not found
no mmc device at slot 2
eMMC or SD Card not detected on mmchost 2

**Reply by xuguosheng110** [reply]
dear you can follow this link https://developer.d-robotics.cc/forumDetail/251934743552436364

**Comment by yuan22442** [comment]
你好，RDK X5中配置有陀螺仪吗

**Comment by xuguosheng110** [comment]
您好，感谢您的来信，我会及时处理！
